File: src/main/resources/scala-native/kf5/KI18n/KLocalizedString.cpp
#include <KLocalizedString>
#include "KLocalizedString.hpp"
#include "../../qt/core/QByteArray.hpp"

void k_localized_string_set_application_domain(q_byte_array domain) {
  KLocalizedString::setApplicationDomain(*((QByteArray*) domain));
  return;
}

File: src/main/scala/kf5/ki18n/KLocalizedString.scala
package kf5.ki18n

import scalanative.unsafe._
import qt.core.QByteArray
import scala.util.Using.Manager

@extern
@link("KF5I18n")
private object _KLocalizedString:
  def k_localized_string_set_application_domain(domain: QByteArray): Unit =
    extern

object KLocalizedString:
  def setApplicationDomain(domain: String)(using m: Manager) =
      _KLocalizedString.k_localized_string_set_application_domain(
        QByteArray(domain)
      )